src/js/date.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @module date
+ * @description Provides helper functions for working with Dates
+ * @see {@link https://github.com/enketo/openrosa-xpath-evaluator/blob/02bb3c301e1c584db87ffdf736a65ea3d630f8e3/src/date-extensions.js}
+ * @see {@link https://github.com/enketo/enketo-xpathjs/blob/e6f6d3d1395e41066b7d0c142a8b5fbb5a0e8098/src/date-extensions.js}
+ *
+ * Note: this module was introduced to address this comment in
+ * `enketo/openrosa-xpath-evaluator`:
+ *
+ * ```js
+ * // TODO probably shouldn't be changing Date.prototype
+ * ```
+ *
+ * The logic was copied from `enketo/enketo-xpathjs` as that appeared to be the
```

Comment:
As a historical note, `enketo-xpathjs` was the original Enketo XPath evaluator that was in use until earlier this year. It was a JS evaluator with a few extensions including this date one. It was pretty slow. Now `openrosa-xpath-evaluator` uses browsers' evaluators which is much faster. I would expect to see this behavior in the evaluator and am surprised to see it copied here. Did this need to happen because the monkeypatching didn't play nice with the new test libs? Would the idea be to bring this back to `openrosa-xpath-evaluator`?

Reply:
We discussed this in our checkin, but for posterity: the issue was that the monkey-patching was happening implicitly in prior test runs given Jasmine's deterministic test order. We decided that this duplication is a risk, and instead I'll directly import the relevant monkey-patch where needed (which we'll also switch to plain functions later).